app/controllers/api/deploy_groups_controller.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+class Api::DeployGroupsController < Api::BaseController
+  include DeployGroupsCommon
+  before_action :enabled?
+
+  def index
+    scope = current_project.present? ? stage.deploy_groups : deploy_groups
```

Comment:
missing `.sort_by(&:natural_order)` for the stage side ... maybe best to copy that bit and not have a concern ?